app/models/link_set.rb

```diff
@@ -1,42 +1,26 @@
 class LinkSet < ActiveRecord::Base
-  include Replaceable
-  include DefaultAttributes
-  include SymbolizeJSON
-
-  validate :links_are_valid
+  has_many :links
 
-  def links=(links_hash)
-    super(links_hash || {})
-  end
+  include Replaceable
 
-private
-  def self.query_keys
-    [:content_id]
-  end
+  def hashed_links
```

Comment:
I understand now. Maybe something like `#grouped_by_type` would be a better name.

I'd question why this needs to be in the model at all? Isn't this only a presentational concern when the links are sent downstream? We have some presenters for that. That might be a better place for this code.